sys/include/ztimer/xtimer_compat.h

```diff
@@ -160,6 +160,15 @@ static inline void xtimer_set_wakeup(xtimer_t *timer, uint32_t offset,
     ztimer_set_wakeup(ZTIMER_USEC, timer, offset, pid);
 }
 
+static inline int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t us)
+{
+    assert(us <= UINT32_MAX);
+    if (ztimer_mutex_lock_timeout(ZTIMER_USEC, mutex, (uint32_t)us)) {
```

Comment:
not simply 

```suggestion
    return ztimer_mutex_lock_timeout(ZTIMER_USEC, mutex, (uint32_t)us);
```
?

Reply:
Sadly some impedance matching is needed. I'll add a comment.